src/MergeContext.js

```diff
@@ -1064,13 +1225,30 @@ class PullRequest {
         const pr = await GH.getPR(this._prNumber(), waitForMergeable);
         assert(pr.number === this._prNumber());
         this._rawPr = pr;
+        this._commitMessage = new CommitMessage(this._rawPr);
     }
 
     // Whether the commit message configuration remained intact since staging.
-    async _messageIsFresh() {
-        const result = this._prMessage() === this._stagedCommit.message;
+    _messageIsFresh() {
+        if (!this._commitMessage.valid()) {
+            this._log("staged commit message is invalid (and stale)");
```

Comment:
It may not be stale in a direct sense of that word, AFAICT, because our validity criteria may have changed while the previously created message was staged (even though there were no PR changes at all):

```suggestion
            this._log("staged commit message became invalid (and will be treated as stale)");
```